### PR TITLE
vita3k: set fs::path type for string paths and refactoring

### PR DESCRIPTION
**Pull request by bookmist**
avoid not need conversion string->wstring->string for linux and macos 
native logging of fs::path
more correct conversion from std path to boost path 
refactor pathes for shaders compiler, shorten parameters lists 
crypto::hex_string: remove not needed null character as last character in string

### CODE REVIEW

File: vita3k/util/include/util/fs.h [hunk 1]
```diff
@@ -19,9 +19,156 @@
...
+ * \return First path + second path
+ */
+fs::path path_concat(const fs::path &path1, const fs::path &path2);
+
+/**
+ * \brief Check if a directory exists and create the directory and all its parents if they don't exist.
+ * \param path Directory to check
+ * \return True if the directory exists or was created successfully, false otherwise.
+ */
+bool force_directories(const fs::path &path);
```
Comment (Macdu) [inline vita3k/util/include/util/fs.h:74]: fs::create_directories already does exactly this (the call to fs::exists before is useless)

Comment (bookmist) [reply vita3k/util/include/util/fs.h:74]: I've seen such code (`if (!fs::exists(dir)) create_directories(dir);`) If it's ok to just call `create_directories(dir);` it's great

File: vita3k/interface.cpp [hunk 2]
```diff
@@ -124,15 +124,15 @@ static bool set_content_path(EmuEnvState &emuenv, const bool is_theme, fs::path
...
 }
 
-bool install_archive_content(EmuEnvState &emuenv, GuiState *gui, const fs::path &archive_path, const ZipPtr &zip, const std::string &content_path, const std::function<void(ArchiveContents)> &progress_callback) {
+bool install_archive_content(EmuEnvState &emuenv, GuiState *gui, const ZipPtr &zip, const std::string &content_path, const std::function<void(ArchiveContents)> &progress_callback) {
     std::string sfo_path = "sce_sys/param.sfo";
     std::string theme_path = "theme.xml";
     vfs::FileBuffer buffer, theme;
 
-    const auto is_theme = mz_zip_reader_extract_file_to_callback(zip.get(), (fs::path(content_path) / theme_path).string().c_str(), &write_to_buffer, &theme, 0);
+    const auto is_theme = mz_zip_reader_extract_file_to_callback(zip.get(), (content_path + "/" + theme_path).c_str(), &write_to_buffer, &theme, 0);
```
[the remark this answers is not in the document]
Comment (bookmist) [reply vita3k/interface.cpp:132]: We get `content_path` from `mz_zip_reader_file_stat` and put it back into `mz_zip_reader_extract_file_to_callback` I think character encoding here is not important at all. (and also path.string() does not convert to utf-8 on old windows (it convert to application (or system) ANSI codepage))

[the remark this answers is not in the document]
Comment (bookmist) [reply vita3k/interface.cpp:132]: But you are right, installing from zip needs re-checking

[the remark this answers is not in the document]
Comment (Macdu) [reply vita3k/interface.cpp:132]: My bad, I didn't notice you were not using a fs path for this specific case

File: vita3k/util/include/util/fs.h (hunk 1 above)
Comment (Macdu) [reply vita3k/util/include/util/fs.h:74]: On the documentation of `create_directories(dir);` for both std and boost, it is written that this function has no effect if the directory exists. The only case where force_directories would behave differently compared to just calling create_directories is if dir already exists but as a file, create_directories would throw an exception while force_directories will do nothing, I am fine with the exception as this defintely wouldn't be expected.

Comment (bookmist) [reply vita3k/util/include/util/fs.h]: Great. Thank you. I replaced force_directories to create_directories
(Also I checked zip unpacking and now it's ready to review)

**Review by Macdu — APPROVED** [review]
Really nice!

**Pull request merged**